### PR TITLE
#137 - Enhancement/implementing populate

### PR DESCRIPTION
**Pull request by lucianopf**
Issue ref: #137

# Reason
Since this is heavily inspired at Mongoose I think we should also implement the populate methods inspired by them.

This is their ref: [Official docs](http://mongoosejs.com/docs/populate.html)

Also that feature makes coding way faster without the need of implementing consecutive queries to get the full result.

# Proposal
Since I'm a new contributor I don't understand every aspect of this library yet, so I'll purpose small and incremental changes until we have a very similar structure of mongoose's populate method. 

By that I'll list goals here so we can keep track and if anyone else wants to get involved follow up the development.

- [x] Implement the populate method
- [x] Attach the method to the Model definition
- [x] Create tests to test this initial feature
- [ ] Implement the ObjectId + ref options in an attribute
- [ ] Implement a better API, such as `Dog.get(1).populate('father')`
- [ ] Implement a better API, such as `Dog.get(1).populate('father.father')`
- [ ] Improve the documentation

### CODE REVIEW

**Comment by lucianopf** [comment]
🤔 

```
nvm install 5 
npm test

143 passing (8s)
Done.
```

Passing with node versions:
- 6.10.3
- 5.12.0
- 4.8.3
- 0.12.18

Something seems wrong at Travis tests

**Comment by brandongoode** [comment]
Just a thought that I wanted to capture before I forgot -- Need to check to make sure it works for a Schema with hashkey.  I think it will, but want to double check.

**Comment by brandongoode** [comment]
One thing I would like to change is the need to manually prepend the model name with the prefix.

File: lib/Model.js [hunk 1]
```diff
@@ -398,6 +406,40 @@ Model.get = function(NewModel, key, options, next) {
...
+  var deferred = Q.defer();
+  var self = this;
+  var Models = this.$__.table.base.models;
+  if (!resultObj) {
+    resultObj = self;
+  }
+  if (!options.path || !options.model) {
+    throw new Error('Invalid parameters provided to the populate method');
+  }
+  try {
```
Comment (brandongoode) [inline lib/Model.js]: Loop through models and compare `Model.$__.name === Model.$__.options.prefix + options.model` -- or somethink like that

Comment (lucianopf) [inline lib/Model.js]: Awesome!!!

That'll avoid manually prefixing such as `'test-Dog6'`

I'll do it on the weekend

**Comment by lucianopf** [comment]
Now it'll accept both the name of the Model or it's Prefix-ModelName

**Comment by brandongoode** [comment]
Looks great.  Thanks

**Pull request merged**